Apps/WebClient/src/Server/Controllers/DependentController.cs

```diff
@@ -0,0 +1,82 @@
+//  limitations under the License.
+// -------------------------------------------------------------------------
+namespace HealthGateway.WebClient.Controllers
+{
+    using System.Security.Claims;
+    using HealthGateway.Common.AccessManagement.Authorization.Policy;
+    using HealthGateway.Common.Models;
+    using HealthGateway.Database.Constants;
+    using HealthGateway.Database.Models;
+    using HealthGateway.Database.Wrapper;
+    using HealthGateway.WebClient.Models;
+    using HealthGateway.WebClient.Services;
+    using Microsoft.AspNetCore.Authorization;
+    using Microsoft.AspNetCore.Http;
+    using Microsoft.AspNetCore.Mvc;
+
+    /// <summary>
+    /// Web API to handle dependent interactions.
+    /// </summary>
+    [Authorize]
+    [ApiVersion("1.0")]
+    [Route("v{version:apiVersion}/api/[controller]")]
+    [ApiController]
+    public class DependentController
+    {
+        private readonly IDependentService dependentService;
+        private readonly IHttpContextAccessor httpContextAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DependentController"/> class.
+        /// </summary>
+        /// <param name="dependentService">The injected user feedback service.</param>
+        /// <param name="httpContextAccessor">The injected http context accessor provider.</param>
+        public DependentController(
+            IDependentService dependentService,
+            IHttpContextAccessor httpContextAccessor)
+        {
+            this.dependentService = dependentService;
+            this.httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <summary>
+        /// Posts a dependent json to be inserted into the database.
+        /// </summary>
+        /// <returns>The http status.</returns>
+        /// <param name="dependent">The user feedback model.</param>
+        /// <response code="200">The dependent record was saved.</response>
+        /// <response code="400">The dependent object is invalid.</response>
+        /// <response code="409">The dependent was already inserted.</response>
+        /// <response code="401">The client must authenticate itself to get the requested response.</response>
+        /// <response code="403">The client does not have access rights to the content; that is, it is unauthorized, so the server is refusing to give the requested resource. Unlike 401, the client's identity is known to the server.</response>
+        [HttpPost]
+        [Authorize(Policy = UserPolicy.UserOnly)]
+        public IActionResult CreateDependent([FromBody] DependentModel dependent)
+        {
+            if (dependent == null)
+            {
+                return new BadRequestResult();
+            }
+            else
+            {
+                ClaimsPrincipal user = this.httpContextAccessor.HttpContext.User;
+                dependent.ParentHdId = user.FindFirst("hdid").Value;
```

Comment:
not sure this is necessary. The parent hdid can be populated already on the request

Reply:
Instead it could validate that parent hdid == user.hdid

Reply:
right, no need, just remove it then